Doppler.Sap.Test/Controllers/BillingControllerTest.cs

```diff
@@ -0,0 +1,33 @@
+using Doppler.Sap.Controllers;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Moq;
+using System;
+using System.Net;
+using Xunit;
+
+namespace Doppler.Sap.Test.Controllers
+{
+    public class BillingControllerTest
+    {
+        private readonly Mock<ILogger<BillingController>> _loggerMock;
+
+        public BillingControllerTest()
+        {
+            _loggerMock = new Mock<ILogger<BillingController>>();
```

Comment:
It is also arguable, so, I have a strong preference but I have not a strong position about how it should be done in this project.

In general, I prefer having all the code and test state inside the test method.

In this case, it would be something like:

```csharp
        public void SetCurrencyRate_ShouldBeHttpStatusCodeOk_WhenCurrencyRateValid()
        {
            var loggerMock = new Mock<ILogger<BillingController>>();
            var controller = new BillingController(loggerMock.Object);
```

But, I know that most of the people in Making think differently than me at this point.